builder/chroots.go

```diff
@@ -0,0 +1,637 @@
+	}
+
+	// Write INI files. These are used to communicate to the next step of mixing (build update).
+	var serverINI bytes.Buffer
+	fmt.Fprintf(&serverINI, `[Server]
+emptydir=%s/empty
+imagebase=%s/image/
+outputdir=%s/www/
+`, stateDir, stateDir, stateDir)
+	if cfg.HasServerSection {
+		fmt.Fprintf(&serverINI, `
+[Debuginfo]
+banned=%s
+lib=%s
+src=%s
+`, cfg.DebugInfoBanned, cfg.DebugInfoLib, cfg.DebugInfoSrc)
+	}
+	err = ioutil.WriteFile(filepath.Join(stateDir, "server.ini"), serverINI.Bytes(), 0644)
+	if err != nil {
+		return err
+	}
+	// TODO: If we are using INI files that are case insensitive, we need to be more restrictive
+	// in bundleset to check for that. See also readGroupsINI in swupd package.
+	var groupsINI bytes.Buffer
+	for _, bundle := range set.Bundles {
+		fmt.Fprintf(&groupsINI, "[%s]\ngroup=%s\n\n", bundle.Name, bundle.Name)
+	}
+	err = ioutil.WriteFile(filepath.Join(stateDir, "groups.ini"), groupsINI.Bytes(), 0644)
+	if err != nil {
+		return err
+	}
+
+	// Mixer is used to create both Clear Linux or a mix of it.
+	var version string
+	if b.Mixver != "" {
+		fmt.Printf("Creating chroots for version %s based on Clear Linux %s\n", b.Mixver, b.Clearver)
+		version = b.Mixver
+	} else {
+		fmt.Printf("Creating chroots for version %s\n", b.Clearver)
+		version = b.Clearver
+	}
+	// TODO: This validation should happen when reading the configuration.
+	if version == "" {
+		return errors.Errorf("no Mixver or Clearver set, unable to proceed")
+	}
+
+	// TODO: Check if this is redundant with the same cleanup in builder.go!
+	chrootVersionDir := filepath.Join(chrootDir, version)
+	fmt.Printf("Removing pre-existing %s before starting\n", chrootVersionDir)
+	err = os.RemoveAll(chrootVersionDir)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("Preparing new %s\n", chrootVersionDir)
+	// TODO: Fix this output when we take bundles from multiple places.
+	fmt.Printf("  based on bundles from: %s\n", b.Bundledir)
+	fmt.Printf("  and yum config: %s\n", b.Yumconf)
+
+	err = os.MkdirAll(chrootVersionDir, 0755)
+	if err != nil {
+		return err
+	}
+	for name, bundle := range set.Bundles {
+		// TODO: Should we embed this information in groups.ini? (Maybe rename it to bundles.ini)
+		var includes bytes.Buffer
+		for _, inc := range bundle.DirectIncludes {
+			fmt.Fprintf(&includes, "%s\n", inc)
+		}
+		err = ioutil.WriteFile(filepath.Join(chrootVersionDir, name+"-includes"), includes.Bytes(), 0644)
+		if err != nil {
+			return err
+		}
+	}
+
+	fmt.Println("Creating os-core chroot")
+	osCoreDir := filepath.Join(chrootVersionDir, "os-core")
+	err = os.MkdirAll(filepath.Join(osCoreDir, "var/lib/rpm"), 0755)
+	if err != nil {
+		return err
+	}
+
+	fmt.Println("Initializing RPM database")
+	_, err = run(
+		"rpm",
+		"--root", osCoreDir,
+		"--initdb",
+	)
+	if err != nil {
+		return err
+	}
+
+	yumCmd := []string{
+		"yum",
+		"--config=" + b.Yumconf,
+		"-y",
+		"--releasever=" + b.Clearver,
+	}
+	// When in Fedora, call dnf instead of yum.
```

Comment:
I know this was in the original version, but we should consider just deprecating yum. Clear Linux is headed in that direction already and dnf is available.

Reply:
Maybe a better question for @bryteise

Reply:
Myself and others have been using dnf without problem for a while AFAIK, let's poke the list and see what people think?

Reply:
+1 @tmarcu

Reply:
Sounds good, but for the sake of this PR, I'll stick to original behavior on this. We later can test and nuke yum usage.

Reply:
Recorded in issue https://github.com/clearlinux/mixer-tools/issues/115.